tools/test/test_depsolve.py

```diff
@@ -219,19 +220,55 @@ def make_dnf_scafolding(base_dir):
     return root_dir, repos_dir, keys_dir
 
 
+def gen_config_combos(items_count):
+    """
+    Generate all possible combinations of indexes of items_count items
+    into two disjoint groups.
+    """
+    indexes = list(range(items_count))
+    all_combinations = []
+
+    for combination_length in range(items_count + 1):
+        for combo_set in combinations(indexes, combination_length):
+            combo_complement_set = tuple(i for i in indexes if i not in combo_set)
+            all_combinations.append((combo_set, combo_complement_set))
+
+    return all_combinations
+
+
+@pytest.mark.parametrize("items_count,expected_combos", (
+    (0, [((), ())]),
+    (1, [
+        ((), (0,)),
+        ((0,), ()),
+    ]),
+    (2, [
+        ((), (0, 1)),
+        ((0,), (1,)),
+        ((1,), (0,)),
+        ((0, 1), ()),
+    ]),
+    (3, [
+        ((), (0, 1, 2)),
+        ((0,), (1, 2)),
+        ((1,), (0, 2)),
+        ((2,), (0, 1)),
+        ((0, 1), (2,)),
+        ((0, 2), (1,)),
+        ((1, 2), (0,)),
+        ((0, 1, 2), ())
```

Comment:
I don't have suggestions but this was hard to read 🙈

Reply:
Yeah, I'm sorry. This is however what was used originally as well. when we hardcoded just the combinations for 2 item repo list... :man_shrugging:

Reply:
That's on me.  IMO this is hard to read but better than the previous implementation.  And the generator is tested so it's even better.